pkg/operator/configobservation/images/observe_images.go

```diff
@@ -148,6 +148,46 @@ func ObserveAllowedRegistriesForImport(genericListers configobserver.Listers, re
 	return observedConfig, errs
 }
 
+func ObserveImagestreamImportMode(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
+	imageStreamImportModePath := []string{"imagePolicyConfig", "imageStreamImportMode"}
+	currentImageStreamImportMode, _, err := unstructured.NestedString(existingConfig, imageStreamImportModePath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentImageStreamImportMode) > 0 {
+		err := unstructured.SetNestedField(prevObservedConfig, currentImageStreamImportMode, imageStreamImportModePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	// now gather the cluster config and turn it into the observed config
+	observedConfig := map[string]interface{}{}
+	configImage, err := listers.ImageConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warningf("image.config.openshift.io/cluster: not found")
+		return observedConfig, errs
```

Comment:
can you please, perhaps with a comment, clarify why we have to special case `NotFound` here, and return `observedConfig` (empty) instead of `prevObservedConfig`?

Reply:
honestly, i followed the pattern of the other observer functions in this file. Going back, it looks like https://github.com/openshift/cluster-openshift-apiserver-operator/pull/57 introduced it. That is a good question but I don't know what the motivation for that was.

Reply:
maybe because if the object is not found, it has been potentially removed and in that case we want it to be empty?

Reply:
probably yes, @sanchezl if the `cluster` object (the thing we are observing for config) is missing, is it an error, or the operator does its best to continue with the old configuration?